src/slurm_cloud_bursting/suspend.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+# the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from logging.config import fileConfig
+
+import argparse
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from configparser import ConfigParser
+from retrying import retry
+
+from common.schedulers.slurm_commands import get_nodes_info, set_nodes_idle
+from common.utils import grouper
+from slurm_cloud_bursting.utils import CONFIG_FILE_PATH
+
+LOG_CONFIG_FILE = "/opt/parallelcluster/configs/slurm/parallelcluster_suspend_logging.conf"
+log = logging.getLogger(__name__)
+
+
+class SlurmSuspendConfig:
+    DEFAULT_MAX_RETRY = 5
+    DEFAULT_MAX_INSTANCES_BATCH_SIZE = 100
+
+    def __init__(self, config_file_path=None, **kwargs):
+        if config_file_path:
+            self._get_config(config_file_path)
+        else:
```

Comment:
if this is needed only for testing what we can do is to have a test for SlurmSuspendConfig that provides a set of files and makes sure this logic is correct and then use a mock for all other functions that require a SlurmSuspendConfig object.